packages/core/src/components/Form.tsx

```diff
@@ -811,7 +812,7 @@ export default class Form<
           disabled={disabled}
           readonly={readonly}
         />
-        {children ? children : <SubmitButton uiSchema={uiSchema} registry={registry} />}
+        {children ? children : <SubmitButton uiSchema={uiSchema} registry={registry} disabled={disabled} />}
```

Comment:
Rather than going through all of this, why not just update the `uiSchema` that gets passed in directly to the `SubmitButton`? Something like:
```
// Add SUBMIT_BTN_OPTIONS_KEY and UI_OPTIONS_KEY to the import for @rjsf/utils
let { [SUBMIT_BTN_OPTIONS_KEY]: submitOptions = {} } = getUiOptions<T, S, F>(uiSchema);
if (disabled) {
  submitOptions = { ...submitOptions, submitButtonProps: { ...submitOptions.submitButtonProps, disable: true } };
}
const submitUiSchema = { [UI_OPTIONS_KEY]: { [SUBMIT_BTN_OPTIONS_KEY]: submitOptions } };
```

Then pass 
```suggestion
        {children ? children : <SubmitButton uiSchema={submitUiSchema} registry={registry} />}
```

It saves a whole bunch of coding and doesn't require updating all of the `SubmitButton` implementations in all themes and changing any types

Reply:
Yes, I did consider this, and I think you're right. 

I was looking at doing something here: https://github.com/rjsf-team/react-jsonschema-form/blob/8ee7d472fa7ef9fdd79113aeff9e012e6bdc6078/packages/core/src/components/Form.tsx#L299

What put me off was around the handling of 'what if the uiSchema exists, what if doesn't, what if nested properties exist....' and then also you've got that state.schemaUtils stuff, which I wasn't sure about. 

But yeah, I think you're right, much nicer than some kind of cascading change. I'll make that change.

Reply:
Using the `getUiOptions()` function will mitigate lack of a `uiSchema` and the two possibilities of `ui:submitButtonOptions` vs the `"ui:options": { submitButtonOptions }`. SchemaUtils do not deal with the `submitButtonOptions`. Also in the suggestion I gave, it should handle the nested options or not case as well